### PR TITLE
Added spanner-instance module

### PR DESCRIPTION
**Pull request by apichick**
---
**Checklist**

I applicable, I acknowledge that I have:
- [X] Read the [contributing guide](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md)
- [X] Ran `terraform fmt` on all modified files
- [X] Regenerated the relevant README.md files using [`tools/tfdoc.py`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#fabric-tools)
- [X] Made sure all relevant tests pass

### CODE REVIEW

File: modules/spanner-instance/variables.tf
Comment (ludoo) [inline modules/spanner-instance/variables.tf:20]: maybe rename to `limits` and `targets`?

File: modules/spanner-instance/iam.tf
Comment (ludoo) [inline modules/spanner-instance/iam.tf:1]: can you use the same iam variables and main file as the other modules? kms would be a good example, keeping  everything identical and just replacing resource names and attributes which reference the main resources?

this will allow easier maintenance of the IAM interface across modules

**Review by ludoo — CHANGES_REQUESTED** [review]

File: modules/spanner-instance/variables.tf
Comment (apichick) [reply modules/spanner-instance/variables.tf]: done

File: modules/spanner-instance/iam.tf
Comment (apichick) [reply modules/spanner-instance/iam.tf:1]: done, but kms iam file has issues

**Comment by apichick** [comment]
do not approve yet, I will do a change to you can have have instance_create flag like in the kms module

**Comment by ludoo** [comment]
> do not approve yet, I will do a change to you can have have instance_create flag like in the kms module

I'll add a useless comment and approve so you can resolve and merge once you're ready :)

File: modules/spanner-instance/README.md
Comment (ludoo) [inline modules/spanner-instance/README.md:5]: just resolve this once you're ready :)

**Review by ludoo — APPROVED** [review]

File: modules/spanner-instance/README.md
Comment (apichick) [reply modules/spanner-instance/README.md:5]: done

**Pull request merged**
